### PR TITLE
check the canary test result and exit if it failed

### PR DESCRIPTION
**Pull request by oliviassss**
### Description
Check and store the canary test result and exit if it failed.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by M00nF1sh — APPROVED** [review]
/lgtm
/approve

File: scripts/run-e2e-test.sh [hunk 1]
```diff
@@ -162,17 +162,24 @@ wait_until_deployment_ready() {
...
   else
     echo "Invalid IP_FAMILY input, choose from IPv4 or IPv6 only"
   fi
+
+  if [[ "$TEST_RESULT" == fail ]]; then
+      echo "e2e test failed."
+      exit 1
+  fi
+  echo "e2e tests completed successfully!"
 }
```
Comment (jdn5126) [inline scripts/run-e2e-test.sh:183]: This needs to go below the call to `run_ginkgo_test`. We also need to make sure we still do the proper cleanup if this fails. So it probably makes sense to add this check last, i.e. right before the "Successfully finished ..."

Comment (oliviassss) [reply scripts/run-e2e-test.sh]: Thanks Jeff. I have moved it to after the tester fetches logs and cleans up the resources

**Review by jdn5126 — APPROVED** [review]

**Review by M00nF1sh — APPROVED** [review]
/lgtm

**Pull request merged**